misc/acrn-config/library/launch_cfg_lib.py

```diff
@@ -601,3 +601,19 @@ def get_vuart1_from_scenario(vmid):
     """Get the vmid's  vuart1 base"""
     vuart1 = common.get_vuart_info_id(SCENARIO_INFO_FILE, 1)
     return vuart1[vmid]['base']
+
+
+def pt_devs_check_audio(audio_map, audio_codec_map):
+    """
+    Check the connections about audio/audio_codec pass-through devices
+    If audio_codec is selected as pass-through device, the audio device
+    must to be chosen as pass-through device either.
+    :param audio_map: the dictionary contains vmid and bdf of audio device
+    :param audio_codec_map: the dictionary contains vmid and bdf of audio_codec device
+    """
+    for vmid in list(audio_map.keys()):
```

Comment:
Do all platforms have two devices (audio device and codec) for audio functionality?

Reply:
I didn't check them on all platforms.